src/rviz/properties/property.h

```diff
@@ -132,9 +132,69 @@ class RVIZ_EXPORT Property : public QObject
   Property(const QString& name = QString(),
            const QVariant& default_value = QVariant(),
            const QString& description = QString(),
-           Property* parent = nullptr,
-           const char* changed_slot = nullptr,
-           QObject* receiver = nullptr);
+           Property* parent = nullptr);
+
+  template <typename Func, typename R>
+  Property(const QString& name,
+           const QVariant& default_value,
+           const QString& description,
+           Property* parent,
+           Func&& changed_slot,
+           const R* receiver)
+    : Property(name, default_value, description, parent)
+  {
+    connect(receiver, std::forward<Func>(changed_slot));
+  }
+
+  // this variant is required to allow omitting the receiver argument
+  template <typename Func, typename P>
+  Property(const QString& name,
+           const QVariant& default_value,
+           const QString& description,
+           P* parent,
+           Func&& changed_slot)
+    : Property(name, default_value, description, parent)
+  {
+    connect(parent, std::forward<Func>(changed_slot));
```

Comment:
```suggestion
    connect(this, &Property::changed,  parent, std::forward<Func>(changed_slot));
```
That's much clearer for me as it avoids another layer of indirection

Reply:
This extra function call is required to dispatch to the old-style `char* slot` variant of connect implemented [here](https://github.com/ros-visualization/rviz/pull/1790/files#diff-ea41e84c9ac9bb8b449bea8284147f9f6e63cd62acd37874e1865567cef1c59eR76-R84) to maintain old behavior.
Surely, we could think of another, more specific method name. However, I used `connect`, because it's actually simply an overload of QObject's connect implementing a connection from the `changed` signal.